xchange-okcoin/src/main/java/info/bitrich/xchangestream/okcoin/OkCoinV3StreamingService.java

```diff
@@ -0,0 +1,155 @@
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import info.bitrich.xchangestream.okcoin.dto.WebSocketMessage;
+import info.bitrich.xchangestream.service.netty.JsonNettyStreamingService;
+import info.bitrich.xchangestream.service.netty.StreamingObjectMapperHelper;
+import info.bitrich.xchangestream.service.netty.WebSocketClientHandler;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufInputStream;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.websocketx.BinaryWebSocketFrame;
+import io.netty.handler.codec.http.websocketx.WebSocketClientHandshaker;
+import io.netty.handler.codec.http.websocketx.WebSocketFrame;
+import io.reactivex.Completable;
+import io.reactivex.CompletableSource;
+import io.reactivex.Observable;
+import io.reactivex.disposables.Disposable;
+import org.knowm.xchange.exceptions.ExchangeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+import java.util.zip.Inflater;
+
+public class OkCoinV3StreamingService extends JsonNettyStreamingService {
+
+  private Observable<Long> pingPongSrc = Observable.interval(15, 15, TimeUnit.SECONDS);
+
+  private Disposable pingPongSubscription;
+
+  public OkCoinV3StreamingService(String apiUrl) {
+    super(apiUrl);
+  }
+
+  @Override
+  public Completable connect() {
+    Completable conn = super.connect();
+    return conn.andThen(
+        (CompletableSource)
+            (completable) -> {
+              try {
+                if (pingPongSubscription != null && !pingPongSubscription.isDisposed()) {
+                  pingPongSubscription.dispose();
+                }
+                pingPongSubscription =
+                    pingPongSrc.subscribe(
+                        o -> {
+                          this.sendMessage("ping");
+                        });
+                completable.onComplete();
+              } catch (Exception e) {
+                completable.onError(e);
+              }
+            });
+  }
+
+  @Override
+  protected String getChannelNameFromMessage(JsonNode message) throws IOException {
+    JsonNode table = message.get("table");
+    if (table != null) {
+      JsonNode data = message.get("data");
+      JsonNode instrument_id = data.get(0).get("instrument_id");
+      String pair = instrument_id.toString();
+      String result = table + ":" + pair;
+      return result.replace("\"", "");
+    }
+    String event = message.get("event").asText();
+    return event;
+  }
+
+  @Override
+  public String getSubscribeMessage(String channelName, Object... args) throws IOException {
+
+    return String.format("{\"op\": \"subscribe\", \"args\": [\"%s\"]}", channelName);
+  }
+
+  @Override
+  public String getUnsubscribeMessage(String channelName) throws IOException {
+    return String.format("{\"op\": \"subscribe\", \"args\": [\"%s\"]}", channelName);
```

Comment:
nit: probably safer to create a `V3WebSocketMessage` class and use `ObjectMapper` to serialise it. (similarly to `OkCoinStreamingService`). Manually building JSON by concatenation can catch you out unexpectedly.  I think it's probably safe here but not good practice.